File: fl_location/android/src/main/kotlin/com/pravera/fl_location/models/LocationAccuracy.kt
package com.pravera.fl_location.models

import com.google.android.gms.location.LocationRequest

enum class LocationAccuracy {
	POWER_SAVE,
	LOW,
	BALANCED,
	HIGH,
	BEST,
	NAVIGATION;

	companion object {
		fun fromIndex(index: Int): LocationAccuracy {
			return when (index) {
				1 -> LOW
				2 -> BALANCED
				3 -> HIGH
				4 -> BEST
				5 -> NAVIGATION
				else -> POWER_SAVE
			}
		}
	}

	fun toPriority(): Int {
		return when (this) {
			POWER_SAVE -> LocationRequest.PRIORITY_LOW_POWER
			LOW -> LocationRequest.PRIORITY_LOW_POWER
			BALANCED -> LocationRequest.PRIORITY_BALANCED_POWER_ACCURACY
			HIGH -> LocationRequest.PRIORITY_HIGH_ACCURACY
			BEST -> LocationRequest.PRIORITY_HIGH_ACCURACY
			NAVIGATION -> LocationRequest.PRIORITY_HIGH_ACCURACY
		}
	}
}

File: fl_location_platform_interface/lib/src/models/location_accuracy.dart
/// An enumeration of location accuracy.
enum LocationAccuracy {
  /// The location has an accuracy of 10km on Android and 3km on iOS.
  powerSave,

  /// The location has an accuracy of 10km on Android and 1km on iOS.
  low,

  /// The location has an accuracy of 100m for both Android and iOS.
  balanced,

  /// The location has an accuracy of ~100m on Android and 10m on iOS.
  high,

  /// The location has an accuracy of ~100m on Android and ~10m on iOS.
  best,

  /// The location has an accuracy of ~100m on Android and optimized for navigation on iOS.
  navigation;

  static LocationAccuracy fromIndex(int? index) =>
      LocationAccuracy.values[index ?? LocationAccuracy.powerSave.index];
}
